### PR TITLE
Fail2ban restore

### LINKED ISSUE #172: Restore Freshrss on a clean server break fail2ban
### Describe the bug

After restoring a full backup of yunohost with freshrss app installed, fail2ban don't start.

### Context

- Hardware: *dedicated server*
- YunoHost version: 11.2.5
- I have access to my server: *Through SSH | through the webadmin*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *no*
- Using, or trying to install package version/branch:1.22.1~ynh1

### Steps to reproduce

- On a yunohost with freshrss, create a full backup
    ```sh
    sudo yunohost backup create
    ```
- On a clean server (other server), restore the backup instead of post install
    ```sh
    sudo yunohost backup restore <name of backup>
    ```
- after the backup is restored, fail2ban failed to run with error :
  - fail2ban                [6382]: ERROR   Failed during configuration: Have not found any log file for freshrss jail
  - fail2ban                [6382]: ERROR   Async configuration of server failed

### Fix the problem
To fix the problem, the file `/var/log/${domaine}-access.log` should be re-created (or restored)

### Expected behavior

The freshrss app shouldn't break the fail2ban service.
The file `/var/log/${domaine}-access.log` should be available

**What is the best solution ?**
- add the file`/var/log/${domaine}-access.log` in backup ?
- create the file if missing during restore ?

### PR DESCRIPTION
**Pull request by gwen-lg**
## Problem

- fail2ban was broken after restoring this app on a clean server
- fix #172 

## Solution

- backup freshrss fail2ban log file and restore it
- recreate the file during restore if needed (missing from the backup)

## PR Status

- [X] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

Note: I have tested than creating the file manually fix the problem

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by gwen-lg** [comment]
This is related to Issue #172

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
Looks good to me. If CI is happy, let's merge :)

**Comment by gwen-lg** [comment]
> I think there is an error on the logpath. I think it should be : `/var/log/nginx/domain-access.log`

I think too, I have created the issue #174 related to this.
Perhaps it's better to fix the logpath first ?

**Comment by gwen-lg** [comment]
But with the path : /var/log/nginx/domain-access.log
If I have several application with use the same domain, it's ok for fail2ban to have several times the same logpath referenced ?
Wouldn't it be more suitable to have the fail2ban configuration linked to the domain, rather than to the application?

**Review by lapineige — APPROVED** [review]
Not a big fan of creating yet another alias, but one might say it avoid typos and not changing all of them correctly.

**Comment by lapineige** [comment]
!testme

File: scripts/restore [hunk 1]
```diff
@@ -69,6 +69,14 @@ ynh_restore_file --origin_path="/etc/logrotate.d/$app"
 #=================================================
 # RESTORE THE FAIL2BAN CONFIGURATION
 #=================================================
+ynh_script_progression --message="Restoring fail2ban configuration..."
+
+ynh_restore_file --origin_path="/var/log/$domain-access.log"
+# If it doesn't exist, create the logfile, required before configuring fail2ban
+log4fail2ban="/var/log/nginx/${domain}-access.log"
+if [ ! -f "${log4fail2ban}" ]; then
```
Comment (ericgaspar) [inline scripts/restore:77]:
```suggestion
if [ ! -f "${log4fail2ban:-}" ]; then
```

File: scripts/upgrade [hunk 2]
```diff
@@ -106,12 +106,13 @@ ynh_use_logrotate --non-append
 ynh_script_progression --message="Upgrading fail2ban configuration..."
 
 # If it doesn't exist, create the logfile, required before configuring fail2ban
-if [ ! -f "/var/log/${domain}-access.log" ]; then
-	touch "/var/log/${domain}-access.log"
+log4fail2ban="/var/log/nginx/${domain}-access.log"
+if [ ! -f "${log4fail2ban}" ]; then
```
Comment (ericgaspar) [inline scripts/upgrade:110]:
```suggestion
if [ ! -f "${log4fail2ban:-}" ]; then
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: scripts/restore (hunk 1 above)
Comment (gwen-lg) [reply scripts/restore:77]: What is the point of `:-` ?
I don't know this `pattern`

**Comment by gwen-lg** [comment]
I have updated the commit, to fix the path according to confirmation of @plopoyop

**Comment by lapineige** [comment]
!testme

File: scripts/restore (hunk 1 above)
Comment (ericgaspar) [reply scripts/restore:77]: cf bash man page -> https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Comment (gwen-lg) [reply scripts/restore:77]: Ok, thank's

**Comment by OniriCorpe** [comment]
see: https://github.com/YunoHost-Apps/freshrss_ynh/pull/173#issuecomment-1868586515

& superseeded by: https://github.com/YunoHost-Apps/freshrss_ynh/pull/179

**Pull request closed without merging**